nupic/encoders/scalar.py

```diff
@@ -253,8 +254,8 @@ def _initEncoder(self, w, minval, maxval, n, radius, resolution):
     """ (helper function)  There are three different ways of thinking about the representation.
      Handle each case here."""
     if n != 0:
-      assert radius == 0
-      assert resolution == 0
+      if (radius !=0 or resolution != 0):
+        raise Exception("ScalarEncoder: n is specified, so radius and resolution should be default")
```

Comment:
This, and other exceptions related to checking the value should be [ValueError](https://docs.python.org/2/library/exceptions.html#exceptions.ValueError) exceptions.